api/envoy/extensions/matching/string/lua/v3/lua.proto

```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+
+package envoy.extensions.matching.string.lua.v3;
```

Comment:
FWIW the `envoy.extensions.matching` namespace is today used for the generic matcher input matchers. These are a very specific kind of match use case. I might be tempted to put this in `envoy.extensions.string_matcher`, but not that opinionated if we want to expand the matching hierarchy use cases. CC @envoyproxy/api-shepherds